src/modules/common/popup/popup.component.js

```diff
@@ -15,161 +18,110 @@ const MODAL_SIZE_CLASS = {
             return null;
         }
 
         return (
-            <h3 className="modal-title c-popup__headline" title={this.props.title}>
-                <span className="c-popup__title">{this.props.title}</span>
-                {this.renderSubtitle()}
+            <h3 className="modal-title c-popup__headline" title={title}>
+                <span className="c-popup__title">{title}</span>
+                {renderSubtitle()}
             </h3>
         );
-    }
-
-    renderSubtitle() {
-        const { subtitle } = this.props;
-
+    };
+    const renderSubtitle = () => {
         if (!subtitle) {
             return null;
         }
 
         return <span className="c-popup__subtitle">{subtitle}</span>;
-    }
-
-    renderFooter() {
-        const { footerChildren } = this.props;
-
+    };
+    const renderFooter = () => {
         if (!footerChildren) {
             return;
         }
 
         return <div className={'modal-footer c-popup__footer'}>{footerChildren}</div>;
-    }
-
-    render() {
-        const { isVisible } = this.state;
-        const { additionalClasses, size, noHeader } = this.props;
-        const modalAttrs = {
-            className: 'c-popup modal fade',
-            ref: this.setModalRef,
-            tabIndex: this.props.hasFocus ? -1 : undefined,
-        };
-
-        document.body.classList.toggle(CLASS_MODAL_OPEN, isVisible);
-        document.body.classList.toggle(CLASS_NON_SCROLLABLE, isVisible);
-
-        if (additionalClasses) {
-            modalAttrs.className = `${modalAttrs.className} ${additionalClasses}`;
-        }
+    };
+    const modalAttrs = {
+        className: classnames({
+            'c-popup modal fade': true,
+            [additionalClasses]: !!additionalClasses,
+            'c-popup--no-header': !!noHeader,
+        }),
+        ref: refModal,
+        tabIndex: hasFocus ? -1 : undefined,
+    };
 
-        if (noHeader) {
-            modalAttrs.className = `${modalAttrs.className} c-popup--no-header`;
-        }
+    useEffect(() => {
+        const modal = $(refModal.current);
 
-        return (
-            <div {...modalAttrs}>
-                <div className={`modal-dialog c-popup__dialog ${MODAL_SIZE_CLASS[size]}`} role="dialog">
-                    <div className="modal-content c-popup__content">
-                        {this.renderHeader()}
-                        <div className="modal-body c-popup__body">{this.props.children}</div>
-                        {this.renderFooter()}
-                    </div>
+        modal.modal({ ...MODAL_CONFIG, isVisible, focus: hasFocus });
+
+        if (isVisible) {
+            document.body.classList.add(CLASS_MODAL_OPEN);
+            document.body.classList.add(CLASS_NON_SCROLLABLE);
+
+            modal.on('keyup', onKeyUp);
+            modal.one('hidden.bs.modal', onClose);
+        } else {
+            modal.modal('hide');
+
+            window.document.body.classList.remove(CLASS_MODAL_OPEN, CLASS_NON_SCROLLABLE);
```

Comment:
just `document`

Reply:
I follow the convention we agreed some time ago when implementing self-invoking functions on load:
```
(function (global, doc) {})(window, window.document);
```